src/Abp.EntityFramework/EntityFramework/Extensions/QueryableExtensions.cs

```diff
@@ -48,5 +56,56 @@ public static IQueryable<T> IncludeIf<T>(this IQueryable<T> source, bool conditi
                 ? source.Include(path)
                 : source;
         }
+
+        /// <summary>
+        /// Nolockings the specified queryable.
+        /// </summary>
+        /// <typeparam name="TEntity">The type of the entity.</typeparam>
+        /// <typeparam name="TResult">The type of the result.</typeparam>
+        /// <param name="repository">The repository.</param>
+        /// <param name="queryable">The queryable.</param>
+        /// <returns></returns>
+        public static TResult Nolocking<TEntity, TResult>(this IRepository<TEntity, int> repository, [NotNull] Func<IQueryable<TEntity>, TResult> queryable) where TEntity : class, IEntity<int>
+        {
+            Check.NotNull(queryable, nameof(queryable));
+
+            TResult result;
+
+            using (var nolockInterceptor = repository.As<AbpRepositoryBase<TEntity, int>>().IocResolver.ResolveAsDisposable<WithNoLockInterceptor>())
+            {
+                using (nolockInterceptor.Object.UseNolocking())
+                {
+                    result = queryable(repository.GetAll());
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Nolockings the specified queryable.
+        /// </summary>
+        /// <typeparam name="TEntity">The type of the entity.</typeparam>
+        /// <typeparam name="TPrimaryKey">The type of the primary key.</typeparam>
+        /// <typeparam name="TResult">The type of the result.</typeparam>
+        /// <param name="repository">The repository.</param>
+        /// <param name="queryable">The queryable.</param>
+        /// <returns></returns>
+        public static TResult Nolocking<TEntity, TPrimaryKey, TResult>(this IRepository<TEntity, TPrimaryKey> repository, [NotNull] Func<IQueryable<TEntity>, TResult> queryable) where TEntity : class, IEntity<TPrimaryKey>
+        {
+            Check.NotNull(queryable, nameof(queryable));
+
+            TResult result;
+
+            using (var nolockInterceptor = repository.As<AbpRepositoryBase<TEntity, TPrimaryKey>>().IocResolver.ResolveAsDisposable<WithNoLockInterceptor>())
+            {
+                using (nolockInterceptor.Object.UseNolocking())
+                {
+                    result = queryable(repository.GetAll());
```

Comment:
@osoykan is that working? I'm asking because Queryable is executed deferred (after ``nolockInterceptor.Object.UseNolocking()`` is disposed). Have you tested this PR with a real database?
Thanks.